ssf/frame.proto

```diff
@@ -0,0 +1,30 @@
+// As we're using required fields, this is proto2 syntax.
+syntax = "proto2";
```

Comment:
It's unclear to me what the support plans are for proto3 and proto2. Is there any danger of proto2 support being missing or going away in the language world?

Since we're going to expect our proto defintitions to work in Stripe's preferred languages and an ecosystem of clients, should we instead just do our own fixed length header? What advantage do we get over doing two fixed length, big-endian numbers as our own header?

Reply:
I think proto2 won't go away anytime soon. We can also define this as proto3, put in some stern warnings & some very comfortable sending functions that do the correct framing automatically. These field defn's will still be fixed-length either way (:

My thinking is that I would like to stick with one transport format (which we have with protobuf already) and use the hell out of this, rather than defining our own. The byte order, versioning, etc is just a bunch of stuff we (and all the implementors of the wire protocol) shouldn't have to think about, IMHO.